lib/sisimai/bite/email/sendgrid.rb

```diff
@@ -0,0 +1,223 @@
+                  # Diagnostic-Code: 550 5.1.1 <userunknown@example.jp>... User Unknown
+                  v['diagnosis'] = cv[1]
+
+                elsif p =~ /\A[Dd]iagnostic-[Cc]ode:[ ]*/ && cv = e.match(/\A[ \t]+(.+)\z/)
+                  # Continued line of the value of Diagnostic-Code header
+                  v['diagnosis'] ||= ''
+                  v['diagnosis']  += ' ' + cv[1]
+                  havepassed[-1] = 'Diagnostic-Code: ' + e
+                end
+              end
+            else
+              # This is an automatically generated message from SendGrid.
+              #
+              # I'm sorry to have to tell you that your message was not able to be
+              # delivered to one of its intended recipients.
+              #
+              # If you require assistance with this, please contact SendGrid support.
+              #
+              # shironekochan:000000:<kijitora@example.jp> : 192.0.2.250 : mx.example.jp:[192.0.2.153] :
+              #   550 5.1.1 <userunknown@cubicroot.jp>... User Unknown  in RCPT TO
+              #
+              # ------------=_1351676802-30315-116783
+              # Content-Type: message/delivery-status
+              # Content-Disposition: inline
+              # Content-Transfer-Encoding: 7bit
+              # Content-Description: Delivery Report
+              #
+              # X-SendGrid-QueueID: 959479146
+              # X-SendGrid-Sender: <bounces+61689-10be-kijitora=example.jp@sendgrid.info>
+              # Arrival-Date: 2012-12-31 23-59-59
+              if cv = e.match(/.+ in (?:End of )?([A-Z]{4}).*\z/)
+                # in RCPT TO, in MAIL FROM, end of DATA
+                commandtxt = cv[1]
+
+              elsif cv = e.match(/\A[Aa]rrival-[Dd]ate:[ ]*(.+)\z/)
+                # Arrival-Date: Wed, 29 Apr 2009 16:03:18 +0900
+                next if connheader['date'].size > 0
+                arrivaldate = cv[1]
+
+                if cv = e.match(/\A[Aa]rrival-[Dd]ate: (\d{4})[-](\d{2})[-](\d{2}) (\d{2})[-](\d{2})[-](\d{2})\z/)
+                  # Arrival-Date: 2011-08-12 01-05-05
+                  arrivaldate += 'Thu, ' + cv[3] + ' '
+                  arrivaldate += Sisimai::DateTime.monthname(0)[cv[2].to_i - 1]
+                  arrivaldate += ' ' + cv[1] + ' ' + [cv[4], cv[5], cv[6]].join(':')
+                  arrivaldate += ' ' + Sisimai::DateTime.abbr2tz('CDT')
+                end
+                connheader['date'] = arrivaldate
+                connvalues += 1
+              end
+            end
+          end
+        end
+
+        return nil if recipients.zero?
+        require 'sisimai/string'
+        require 'sisimai/smtp/status'
+
+        dscontents.map do |e|
+          e['diagnosis'] = Sisimai::String.sweep(e['diagnosis'])
+
+          # Get the value of SMTP status code as a pseudo D.S.N.
+          if cv = e['diagnosis'].match(/\b([45])\d\d[ \t]*/)
+            # 4xx or 5xx
+            e['status'] = sprintf("%d.0.0", cv[1])
```

Comment:
__[rubocop]__
* Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

<!-- Comment by [SideCI](https://sideci.com) -->